openmdao/visualization/n2_viewer/n2_viewer.py

```diff
@@ -378,7 +381,28 @@ def _get_viewer_data(data_source):
             return {}
 
     elif isinstance(data_source, str):
-        data_dict = CaseReader(data_source, pre_load=False).problem_metadata
+        cr = CaseReader(data_source, pre_load=True)
+
+        data_dict = cr.problem_metadata
+
+        if case_id is not None:
+            if isinstance(case_id, str):
+                if 'root' in case_id:
+                    raise ValueError(f"case_id is not a driver case. Find valid case_id with "
+                                     f"om.CaseReader('{data_source}').list_cases()")
+                sys_cases = cr._driver_cases._cases[case_id]
+            else:
+                cases = [key for key in cr._driver_cases._cases.keys()]
+                sys_cases = cr._driver_cases._cases[cases[case_id]]
```

Comment:
a nit maybe, but why is this called `sys_cases` when it's actually driver cases.. maybe just `cases` would be less misleading...
also, I think it might be better to stick to the public CaseReader API rather than accessing private attributes